apiclient/thread.go

```diff
@@ -82,3 +83,21 @@ func (c *Client) ListThreads(ctx context.Context, opts ListThreadsOptions) (resu
 	_, err = toObject(resp, &result)
 	return
 }
+
+func (c *Client) UploadThreadFile(ctx context.Context, projectID string, threadID string, assistantID string, filename string, fileContent []byte) error {
+	_, resp, err := c.doRequest(ctx, http.MethodPost, fmt.Sprintf("/assistants/%s/projects/%s/threads/%s/files/%s", assistantID, projectID, threadID, filename), bytes.NewReader(fileContent))
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	return nil
```

Comment:
Does this also work?

```suggestion
	defer resp.Body.Close()
	return err
```

Just a bit more concise if it does. This could also be changed in the other function in this PR.

Reply:
yeah I think it works, resp.body should always be non-nil

Reply:
I don't think this works because `resp` is probably `nil` when a non-nil error is returned.

Reply:
Oh. donnie is correct